tests/vm/tcompilesetting.nim

```diff
@@ -1,9 +1,9 @@
 discard """
-cmd: "nim c --nimcache:build/myNimCache --nimblePath:myNimblePath $file"
+matrix: "--nimcache:build/myNimCache --nimblePath:myNimblePath --passc:'-fmax-errors=4' --passl:'-u dummysymboldoesnotexist'"
```

Comment:
```suggestion
matrix: "--nimcache:build/myNimCache --nimblePath:myNimblePath --passc:'-fmax-errors=4' --passl:'-Xlinker --defsym -Xlinker dummysymboldoesnotexist=0x0'"
```
I cannot test it with the XCode toolchain, but I think that this should work.

Reply:
LooooooL, no idea why I thought guarding the compiler query would change the outcome of the test considering the cli params. I'm a bit distracted today. 😮‍💨 

I'll get things building again on my mac (so slow) and see if this works there.